src/components/FunctionsPage/Functions.js

```diff
@@ -50,6 +52,35 @@ const Functions = ({
   const [taggedFunctions, setTaggedFunctions] = useState([])
   const [functionsPanelIsOpen, setFunctionsPanelIsOpen] = useState(false)
   const location = useLocation()
+  let fetchFunctionLogsTimeout = useRef(null)
+
+  const handleFetchFunctionLogs = useCallback(
+    (projectName, name, tag, offset) => {
+      return fetchFunctionLogs(projectName, name, tag, offset).then(result => {
+        if (
+          result.headers?.['x-mlrun-function-status'] === 'pending' ||
+          result.headers?.['x-mlrun-function-status'] === 'running'
```

Comment:
Please declare a `TRANSIENT_FUNCTION_STATUSES` constant which is an array `['pending', 'running']`.
You can even put it in some `util` file related to the functions page.
Then use it like so:

```suggestion
          TRANSIENT_FUNCTION_STATUSES.includes(result.headers?.['x-mlrun-function-status'])
```

This way the list of these statuses is much more maintainable.
It is stored in a single place and it is easy to add/remove to/from this list in the future if/when needed, without having to change a single character of code here.